src/backend/distributed/utils/node_metadata.c

```diff
@@ -455,6 +455,7 @@ ActivateNode(char *nodeName, int nodePort)
 
 	if (WorkerNodeIsPrimary(workerNode))
 	{
+		ReplicateAllDependenciesToNode(nodeName, nodePort);
```

Comment:
At the start of reference table replication step, we give a NOTICE because it might take a while and we don't want users to cancel it.

Would it make sense to have something similar for dependencies? We could do it if there are, say `> 1000` dependencies that we need to create.

Reply:
yeah, could be nice, I will have a look at finding that notice.
in that case the user might see multiple notices during node activation, any reasons that would not be great?

Also, what is a reasonable cutoff size? even a 1000 was relatively fast iirc

Reply:
> yeah, could be nice, I will have a look at finding that notice.

https://github.com/citusdata/citus/blob/master/src/backend/distributed/utils/reference_table_utils.c#L313

> Also, what is a reasonable cutoff size? even a 1000 was relatively fast iirc

I think the time depends on the shape of the graph and size of pg_depend/pg_dist_object etc. But, with 1000 objects it could take more than 50 seconds, I can show it locally :) So, to be consistent, maybe notice if `> 1`?

Reply:
Depends what is taking the time, if it is already taking a long time while walking the graph we don't even know the size yet.

Can you show me tomorrow, we might be able to flamegraph it to get an understanding where the time is used.

Reply:
The notice is implemented in the body of `ReplicateAllDependenciesToNode`.
I have chosen an arbitrary `100` as a cutoff to show the notice.